test/test_cgi_server.rb

```diff
@@ -0,0 +1,140 @@
+# coding: utf-8
+# frozen_string_literal: false
+
+require 'test/unit'
+require 'webrick'
+require_relative 'webrick_testing'
+require 'tempfile'
+require "xmlrpc/server"
+require 'xmlrpc/client'
+
+class Test_CGIServer < Test::Unit::TestCase
+  include WEBrick_Testing
+
+  def setup_http_server_option(use_ssl)
+    option = {
+      :BindAddress => "localhost",
+      :Port => 0,
+      :SSLEnable => use_ssl,
+    }
+    if use_ssl
+      require 'webrick/https'
+      option.update(
+        :SSLVerifyClient => ::OpenSSL::SSL::VERIFY_NONE,
+        :SSLCertName => []
+      )
+    end
+
+    option
+  end
+
+  def test_client_server
+    omit("The CGI file does not work on Windows") if RUBY_PLATFORM =~ /(mswin|mingw)/
+
+    # NOTE: I don't enable SSL testing as this hangs
+    Tempfile.create("cgi-bin") do |tempfile|
+      tempfile.write(cgi_bin_script)
+      tempfile.close
+      File.chmod(0755, tempfile.path)
+
+      [false].each do |use_ssl|
+        option = setup_http_server_option(use_ssl)
+        with_server(option, WEBrick::HTTPServlet::CGIHandler, tempfile.path) {|addr|
+          @s = XMLRPC::Client.new3(:host => addr.ip_address, :port => addr.ip_port, :use_ssl => use_ssl)
+          @s.user = 'admin'
+          @s.password = 'admin'
+          silent do
+            do_test
+          end
+          @s.http.finish
+          @s = XMLRPC::Client.new3(:host => addr.ip_address, :port => addr.ip_port, :use_ssl => use_ssl)
+          @s.user = '01234567890123456789012345678901234567890123456789012345678901234567890123456789'
+          @s.password = 'guest'
+          silent do
+            do_test
+          end
+          @s.http.finish
+        }
+      end
+    end
+  end
+
+  def silent
+    begin
+      back, $VERBOSE = $VERBOSE, nil
+      yield
+    ensure
+      $VERBOSE = back
+    end
+  end
+
+  def do_test
+    # simple call
+    assert_equal 9, @s.call('test.add', 4, 5)
+
+    # fault exception
+    assert_raise(XMLRPC::FaultException) { @s.call('test.div', 1, 0) }
+
+    # fault exception via call2
+    ok, param = @s.call2('test.div', 1, 0)
+    assert_equal false, ok
+    assert_instance_of XMLRPC::FaultException, param
+    assert_equal 1, param.faultCode
+    assert_equal 'division by zero', param.faultString
+
+    # call2 without fault exception
+    ok, param = @s.call2('test.div', 10, 5)
+    assert_equal true, ok
+    assert_equal param, 2
+
+    # introspection
+    assert_equal ["test.add", "test.div", "system.listMethods", "system.methodSignature", "system.methodHelp"], @s.call("system.listMethods")
```

Comment:
Could you avoid omitting parenthesis for assertions?
See also: https://bugs.ruby-lang.org/issues/20922

Could you align long expected and actual for readability?

```suggestion
    assert_equal(["test.add", "test.div", "system.listMethods", "system.methodSignature", "system.methodHelp"],
                 @s.call("system.listMethods"))
```